fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeMgr.java

```diff
@@ -449,7 +453,62 @@ private void dropPartitionStatistics() {
         }
     }
 
-    private void checkAndDropPartitionStatistics() {
+    private void clearStalePartitionStats() {
+        // It means FE is restarted, the previous step had cleared the stats.
+        if (lastCleanTime == null) {
+            lastCleanTime = LocalDateTime.now();
+            return;
+        }
+
+        //  do the clear task once every 12 hours.
+        if (Duration.between(lastCleanTime, LocalDateTime.now()).toSeconds() < Config.clear_stale_stats_interval_sec) {
+            return;
+        }
+
+        List<Table> tables = Lists.newArrayList();
+        LocalDateTime workTime = LocalDateTime.now();
+        for (Map.Entry<Long, AnalyzeStatus> entry : analyzeStatusMap.entrySet()) {
+            AnalyzeStatus analyzeStatus = entry.getValue();
+            LocalDateTime endTime = analyzeStatus.getEndTime();
+            // After the last cleanup, if a table has successfully undergone a scheduled full statistics collection,
+            // and the collection completion time is after the last cleanup time,
+            // then during the next cleanup process, the stale column statistics would be cleared.
+            if (analyzeStatus instanceof NativeAnalyzeStatus
+                    && analyzeStatus.getScheduleType() ==  StatsConstants.ScheduleType.SCHEDULE
+                    && analyzeStatus.getStatus() == StatsConstants.ScheduleStatus.FINISH
+                    && endTime.isAfter(lastCleanTime)) {
+                NativeAnalyzeStatus nativeAnalyzeStatus = (NativeAnalyzeStatus) analyzeStatus;
+                Database db = GlobalStateMgr.getCurrentState().getDb(nativeAnalyzeStatus.getDbId());
+                if (db != null && db.getTable(nativeAnalyzeStatus.getTableId()) != null) {
+                    tables.add(db.getTable(nativeAnalyzeStatus.getTableId()));
+                }
+            }
+        }
+
+        if (tables.isEmpty()) {
+            lastCleanTime = workTime;
+        }
+
+        List<Long> tableIds = Lists.newArrayList();
+        int exprLimit = 50000;
+        for (int i = 0; i < tables.size(); i++) {
+            tableIds.add(tables.get(i).getId());
+            if (i == tables.size() - 1 || tableIds.size() + 1 == exprLimit) {
+                ConnectContext statsConnectCtx = StatisticUtils.buildConnectContext();
+                statsConnectCtx.setStatisticsConnection(true);
+                statsConnectCtx.getSessionVariable().setExprChildrenLimit(exprLimit * 2);
+                StatisticExecutor executor = new StatisticExecutor();
+                boolean res = executor.clearStaleColumnStatistics(statsConnectCtx, tableIds, lastCleanTime);
```

Comment:
can't clean statistics only by table_id and update_time, because table may only update some partition, it's will drop whole statistics